woocommerce/woocommerce
# Add context to errors

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds a context to the error messages as part of the changes we will need to accomplish the addition of a link to the error snackbar https://github.com/woocommerce/woocommerce/issues/46768

![Screenshot 2024-07-10 at 4 24 59 PM](https://github.com/woocommerce/woocommerce/assets/1314156/91fe4c8d-b24f-4654-b4e9-59e6f3b727e6)


Partially addresses https://github.com/woocommerce/woocommerce/issues/46768.

### ACs
The error snackbar should continue working as usual.
- [ ] When there is a field with an error in a tab that is not visible, the error snackbar won't dismiss automatically and will have a close button (`x`).
- [ ] When there is a field with an error in a tab that is visible, the error snackbar won't have a close button (`x`) and it will dismiss automatically. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure `Try the new product editor (Beta)` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add New.
3. Add a regular price (don't add a product name) and press Publish.
4. A snackbar with the message: `Product name is required.` will be visible.
5. Change tab and press Publish again.
6. A snackbar with the message: `Product name is required.` will be visible. The snackbar will have dismiss button (`x`) and it won't dismiss automatically.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
